File: l12.py
e = {1: "I", 4: "IV", 5: "V", 9: "IX", 10: "X", 40: "XL", 50: "L", 90: "XC", 100: "C", 400: "CD", 500: "D", 900: "CM", 1000: "M"}
k = [1000,900,500,400,100,90,50,40,10,9,5,4,1]
class Solution(object):
	def intToRoman(self, num):
		rt = ""
		while num>0:
			for i in k:
				while num /i >0:
					rt += e[i]
					num -= i
		return rt
print Solution().intToRoman(20)

File: l12.cpp
#include <iostream>
#include <string>
using namespace std;

string intToRoman(int num) {
string rt = "";

string roman[13]={"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
int number[13]={1,4,5,9,10,40,50,90,100,400,500,900,1000};
while (num > 0) {
	for(int i = 12;i>=0;i--){1
		while(num-number[i]>=0){
			rt+=roman[i];
			num-=number[i];
		}
	}
}
return rt;
}

int main()
{
	cout<<intToRoman(499);
}
